### PR TITLE
`Development`: Forward exceptions to sentry during exam monitoring

### PR DESCRIPTION
**Pull request by Dominik-Weinzierl**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

In order to get notified about any issues, we want to forward the exceptions during the exam monitoring to sentry.

### Description

- Added `captureException(error);` in each catch block

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Exam

1. Log in to Artemis
2. Create an Exam
3. Enable Exam Monitoring
4. Participate as Student
5. Verify that everything works as before

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-monitoring.service.ts | 86.66% | 100% |

### CODE REVIEW

File: src/test/javascript/spec/service/exam-monitoring.service.spec.ts [hunk 1]
```diff
@@ -170,4 +173,33 @@ describe('ExamMonitoringService', () => {
...
+        expect(captureExceptionSpy).toHaveBeenCalledOnce();
+        expect(captureExceptionSpy).toHaveBeenCalledWith(error);
+    });
+
+    it.each(createExamActions())('should send exception to sentry during invalid handle', (action: ExamAction) => {
+        const error = new Error('Invalid');
+        const spy = jest.spyOn(artemisServerDateService, 'now').mockImplementation(() => {
+            throw error;
+        });
+        const captureExceptionSpy = jest.spyOn(Sentry, 'captureException');
```
Comment (canberkanar) [inline src/test/javascript/spec/service/exam-monitoring.service.spec.ts:198]: See above.

**Review by canberkanar — DISMISSED** [review]
Code changes look good to me. Just added a little recommendation you may implement on your will.

**Comment by ge65cer** [comment]
Code approved. Have you tested that you can quickly spot errors related to the exam monitoring in Sentry? 
Maybe [custom tags](https://docs.sentry.io/platforms/javascript/enriching-events/tags/) or alike could help with that.

**Review by manuelmanso — APPROVED** [review]
Looks good :)

**Review by maximiliansoelch — APPROVED** [review]
Everything I tested on ts1 regarding exams works as before

**Pull request merged**